services-camel/src/main/java/edu/unc/lib/cdr/FulltextProcessor.java

```diff
@@ -0,0 +1,83 @@
+package edu.unc.lib.cdr;
+
+import static edu.unc.lib.cdr.headers.CdrFcrepoHeaders.CdrBinaryMimeType;
+import static edu.unc.lib.cdr.headers.CdrFcrepoHeaders.CdrBinaryPath;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_URI;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.sax.BodyContentHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+import edu.unc.lib.dl.fcrepo4.BinaryObject;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.Repository;
+import edu.unc.lib.dl.rdf.PcdmUse;
+
+public class FulltextProcessor implements Processor {
+	private static final Logger log = LoggerFactory.getLogger(AddDerivativeProcessor.class);
+	
+	private final Repository repository;
+	private final String slug;
+	private final String fileSuffix;
+
+	public FulltextProcessor(Repository repository, String slug, String fileSuffix) {
+		this.repository = repository;
+		this.slug = slug;
+		this.fileSuffix = fileSuffix;
+	}
+
+	@Override
+	public void process(Exchange exchange) throws Exception {
+		final Message in = exchange.getIn();
+
+		String binaryUri = (String) in.getHeader(FCREPO_URI);
+		String binaryMimeType = (String) in.getHeader(CdrBinaryMimeType);
```

Comment:
Wondering whether we should change those header names to be in ALL_CAPS_SEPARATED_BY_UNDERSCORES to follow Java conventions. I had thought they were variable names, not constants :-)